pkg/templating/manifest_generator.go

```diff
@@ -37,22 +35,54 @@ const (
 	legalCharSeparator = "-"
 )
 
+// Maximum length for UID, as we are already consuming quite a few of the remaining in templates (253-200 = 53)
+const templateUIDMaxLength int = 200
+
+func NewHandler(accessToken, threescaleURL, svcID string, fixup bool) (*Handler, []error) {
+	var errs []error
+
+	// validate that we can create a k8s label with this value
+	newSvcID, validationErrs := K8sLabelValueValidation(svcID, -1, fixup)
+	if len(validationErrs) > 0 {
+		errs = append(errs, validationErrs...)
+	}
+
+	return &Handler {
+		AccessToken: accessToken,
+		SystemURL:   threescaleURL,
+		ServiceID:   newSvcID,
+	}, errs
+}
+
 // NewConfigGenerator - constructor for ConfigGenerator which validates the input and generates a UID
 // Constructor should be used in order to guarantee the generation of valid templates.
-func NewConfigGenerator(h Handler, i Instance, r Rule) (*ConfigGenerator, error) {
-	u, err := h.parseURL()
+func NewConfigGenerator(h Handler, i Instance, r Rule, uid string, fixup bool) (*ConfigGenerator, []error) {
+	var errs []error
+
+	u, err := ParseURL(h.SystemURL)
 	if err != nil {
-		return nil, err
+		errs = append(errs, err)
 	}
 
-	uid, err := h.uidGenerator(u)
-	if err != nil {
-		return nil, fmt.Errorf("UID generation failed: %s", err.Error())
+	// validate that we can create a k8s resource with this name
+	newUID, validationErrs := K8sResourceNameValidation(uid, templateUIDMaxLength, fixup)
+	if len(validationErrs) > 0 {
+		errs = append(errs, validationErrs...)
+	}
+
+	// validate that we can create a k8s label with this value
+	newHostname, validationErrs := K8sLabelValueValidation(u.Hostname(), -1, fixup)
+	if len(validationErrs) > 0 {
+		errs = append(errs, validationErrs...)
+	}
+
+	if len(errs) > 0 {
+		return nil, errs
```

Comment:
I'm struggling to understand the behaviour here. IF the user has asked for this to be fixed up and that has been a success should we still return the errors? Or will that situation occur?